service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/mapper/EncounterMapper.java

```diff
@@ -56,17 +66,11 @@ public Encounter mapEncounter(POCDMT000002UK01ClinicalDocument1 clinicalDocument
         encounter.setPeriod(getPeriod(clinicalDocument));
         setServiceProvider(encounter, clinicalDocument);
         setAppointment(encounter, clinicalDocument);
-        setPatient(encounter, clinicalDocument);
+        setSubject(encounter, clinicalDocument);
         setEpisodeOfCare(encounter, clinicalDocument);
         return encounter;
     }
 
-    private void setPatient(Encounter encounter, POCDMT000002UK01ClinicalDocument1 clinicalDocument) {
-        Patient patient = patientMapper.mapPatient();
-        encounter.setSubjectTarget(patient);
-        encounter.setSubject(new Reference(patient));
-    }
-
     private void setEpisodeOfCare(Encounter encounter, POCDMT000002UK01ClinicalDocument1 clinicalDocument) {
         Optional<EpisodeOfCare> episodeOfCare = episodeOfCareMapper.mapEpisodeOfCare(clinicalDocument, encounter.getSubject());
         if (episodeOfCare.isPresent()) {
```

Comment:
episodeOfCare.ifPresent(ofCare -> encounter.addEpisodeOfCare(new Reference(ofCare))); 
just a suggestion :)

Reply:
Module updated also - actually that was Dawid's code but I agree with you.